app/src/main/java/zapsolutions/zap/customView/WalletSpinner.java

```diff
@@ -0,0 +1,128 @@
+
+                    } else {
+                        // Load the selected wallet
+                        adapterView.getAdapter().getItem(position);
+                        PrefsUtil.edit().putString(PrefsUtil.CURRENT_WALLET_CONFIG, selected).commit();
+
+                        // Do not ask for pin again...
+                        TimeOutUtil.getInstance().restartTimer();
+
+                        Wallet.getInstance().reset();
+
+
+                        // This has to be done, otherwise the OnDestroy method of the old
+                        // HomeActivity will be called after the OnCreate Method of the new one.
+                        // This would result in all listeners etc. to be unsubscribe right after wallet switch.
+                        ((HomeActivity) getContext()).prepareWalletSwitch();
+
+                        // Open home and clear history
+                        Intent intent = new Intent(getContext(), HomeActivity.class);
+                        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TASK | Intent.FLAG_ACTIVITY_NEW_TASK);
+                        getContext().startActivity(intent);
+                    }
+                } else {
+                    if (PrefsUtil.isWalletSetup()) {
+                        // Open the create wallet ui
+                        Intent intent = new Intent(getContext(), SetupActivity.class);
+                        intent.putExtra("setupMode", SetupActivity.ADD_WALLET);
+                        getContext().startActivity(intent);
+                    } else {
+                        // Initiate wallet setup
+                        Intent intent = new Intent(getContext(), SetupActivity.class);
+                        intent.putExtra("setupMode", SetupActivity.FULL_SETUP);
+                        getContext().startActivity(intent);
+                    }
+                    // If going back we don't want to have "Add new..." selected
+                    adapterView.setSelection(0);
+                }
+            }
+
+            @Override
+            public void onNothingSelected(AdapterView<?> adapterView) {
+
+            }
+        });
+    }
+
+    private void updateList() {
+
+        if (PrefsUtil.isWalletSetup()) {
+            String[] items = new String[WalletConfigsManager.getInstance().getAllWalletConfigs(true).size() + 1];
+            for (int i = 0; i < WalletConfigsManager.getInstance().getAllWalletConfigs(true).size(); i++) {
+                items[i] = WalletConfigsManager.getInstance().getAllWalletConfigs(true).get(i).getAlias();
+            }
+            items[items.length - 1] = "Add new...";
+            ArrayAdapter<String> adapter = new ArrayAdapter<>(getContext(), android.R.layout.simple_spinner_dropdown_item, items);
+            //set the spinners adapter to the previously created one.
+            this.setAdapter(adapter);
+        } else {
+            String[] items = new String[2];
```

Comment:
I know this is still work in progress, but for now would it be better to not show the switch until the wallet is setup?
Since user cannot switch until he has multiple wallets, and we don't really have a wallet to show in the spinner.
Also adding a wallet is already prominent with the home screen button.

Reply:
Actually if the spinner will end up in the final design (without having to be enabled in the settings), I would like to already have it in the demo mode. The reason is the demo mode shall give the user the feeling of what the app will be like when it is setup.